.github/workflows/vm-kernel.yaml

```diff
@@ -21,22 +21,42 @@ jobs:
     outputs:
       image: ${{ fromJSON(steps.build-linux-kernel.outputs.metadata)['image.name'] }}@${{ steps.build-linux-kernel.outputs.digest }}
 
-    # Use the biggest GitHub Hosted runner
-    runs-on: gha-ubuntu-22.04-64cores
+    runs-on: [ self-hosted, gen3, large ]
     steps:
-
       - name: git checkout
         uses: actions/checkout@v3
 
+      - name: set custom docker config directory
+        run: |
+          mkdir -p .docker-custom
+          echo DOCKER_CONFIG=$(pwd)/.docker-custom >> $GITHUB_ENV
+      - name: login to docker hub
+        run: |
+          DOCKERHUB_AUTH=$(echo -n "${{ secrets.NEON_DOCKERHUB_USERNAME }}:${{ secrets.NEON_DOCKERHUB_PASSWORD }}" | base64)
+          echo "::add-mask::${DOCKERHUB_AUTH}"
+
+          cat <<-EOF > .docker-custom/config.json
+            {
+              "auths": {
+                "https://index.docker.io/v1/": {
+                  "auth": "${DOCKERHUB_AUTH}"
+                }
+              }
+            }
+          EOF
```

Comment:
I'm not too familiar with github actions semantics, so at first glance, I'd be worried about leaking credentials here - but as long as it doesn't, I have no objections

Reply:
@bayandin I am not worried about the logs file however why do we save the credentials in a config file? Can't we just set it as an environment variable to avoid saving it in a file?

Reply:
I didn't find a way to use an environment variable for docker authorisation. However, I found that we don't need to configure this file manually, we can use `docker/login-action@v2` (as we used before)

A workflow run: https://github.com/neondatabase/autoscaling/actions/runs/7129968930